### PR TITLE
Feature/#137/sign in with existing user

### PR DESCRIPTION
**Pull request by JustTheDude001**
Feature in order to sign in with an existing user.

git diff --stat main | grep +

```

.env.docker                                        |    4 +
 .../SignInUser/SignInUserAnnotations.php           |   89 ++
 app/DTO/UserDTO.php                                |   15 +
 app/Http/Controllers/api/AuthController.php        |   61 +
 app/Http/Requests/SigninRequest.php                |   38 +
 app/Service/User/UserService.php                   |  123 ++
 composer.json                                      |    1 +
 composer.lock                                      |   63 +-
 config/database.php                                |   22 +
 database/seeders/DatabaseSeeder.php                |    1 +
 database/seeders/SigninTestSeeder.php              |   40 +
 docker-compose.yml                                 |   13 +
 docker/redis/.conf/redis.conf                      | 1320 ++++++++++++++++++++
 reboot-linux.bat                                   |    2 +
 reboot.bat                                         |    1 +
 routes/api/v1.php                                  |    4 +-
 .../Feature/Controller/User/SigninMockeryTest.php  |  230 ++++
 tests/Feature/Controller/User/SigninTest.php       |  118 ++
 tests/Feature/Service/User/UserServiceTest.php     |  291 +++++
 402 files changed, 2434 insertions(+), 2 deletions(-)

```
image

image

### CODE REVIEW

**Review by jordimorillo — APPROVED** [review]
There are a few silly things you can fix with unused code, methods and things like that. Get Phpstorm that will highlight all those details and repair them. The PR is functional so congratulations! :)

File: app/Service/User/UserService.php
Comment (Agonpas) [inline app/Service/User/UserService.php:1]: Add declare(strict_types=1);

**Review by Agonpas — APPROVED** [review]
You have done a great job!!!

File: app/Service/User/UserService.php
Comment (JustTheDude001) [reply app/Service/User/UserService.php:1]: Bueno.. te hago un copy y paste del discord que tengo sueño... 
```
Muchas gracias Albert!!!
Me ha ayudado muchisimo lo que me has dicho para depurar el codigo
Incluso he encontrado un error grave en el testing ^^'
si pudieras hacerme el favor de volver a testear la rama con el ultimo commit ya seria idilico 🙂
Muchas gracias de nuevo!
Me has salvado **** haha
```

Comment (JustTheDude001) [reply app/Service/User/UserService.php:1]: image
Resultados actuales del testing

Comment (JustTheDude001) [reply app/Service/User/UserService.php:1]: Gracias de nuevo!

**Pull request merged**